### PR TITLE
Add support for custom headers.

### PR DESCRIPTION
**Pull request by alanpeabody**
@werbitzky Here is my first pass at this. I still need to verify it fulfills my requirements (AWS Auth), but I thought I would open for feedback.

### CODE REVIEW

File: lib/elastix/http.ex [hunk 1]
```diff
@@ -18,20 +18,14 @@ defmodule Elastix.HTTP do
...
-    end
+    full_headers = headers
+                   |> add_content_type_header
+                   |> add_shield_header
+                   |> add_custom_headers(method, full_url, body)
 
     options = Keyword.merge(default_httpoison_options(), options)
-    HTTPoison.Base.request(
+    {m, f, _a} = Elastix.config(:test_request_mfa) || {HTTPoison.Base, :request, []}
+    apply(m, f, [
```
Comment (alanpeabody) [inline lib/elastix/http.ex]: I added this bit so we could hook in and test request data without actually sending them via HTTP.

**Comment by alanpeabody** [comment]
@werbitzky do you have any feedback on this?

**Comment by werbitzky** [comment]
@alanpeabody thanks, I like your solution!

**Comment by werbitzky** [comment]
@alanpeabody do you want to add anything or should I merge?

**Comment by alanpeabody** [comment]
@werbitzky nope, should be good to go. Already using in production. :-D

File: lib/elastix/http.ex [hunk 2]
```diff
@@ -18,20 +18,14 @@ defmodule Elastix.HTTP do
...
-      content_headers
-    end
+    full_headers = headers
+                   |> add_content_type_header
+                   |> add_shield_header
+                   |> add_custom_headers(method, full_url, body)
 
     options = Keyword.merge(default_httpoison_options(), options)
-    HTTPoison.Base.request(
+    {m, f, _a} = Elastix.config(:test_request_mfa) || {HTTPoison.Base, :request, []}
```
Comment (junaid1460) [inline lib/elastix/http.ex:27]: This could be used to override HTTP lib, can we rename this `test_request_mfa` atom?

**Pull request merged**